astronomer/providers/apache/spark/operators/spark_sumbit.py

```diff
@@ -0,0 +1,105 @@
+    :param name: Name of the job (default airflow-spark). (templated)
+    :param num_executors: Number of executors to launch
+    :param application_args: Arguments for the application being submitted (templated)
+    :param env_vars: Environment variables for spark-submit. It supports yarn and k8s mode too. (templated)
+    :param verbose: Whether to pass the verbose flag to spark-submit process for debugging
+    :param spark_binary: The command to use for spark submit.
+                         Some distros may use spark2-submit.
+    """
+
+    def __init__(self, *, poll_interval: int = 5, **kwargs: Any) -> None:
+        self.poll_interval = poll_interval
+        super().__init__(**kwargs)
+
+    def execute(self, context: "Context") -> None:
+        """
+        Airflow runs this method on the worker and defers using the trigger and make a call to
+        SparkSubmitHookAsync to run the provided spark job
+        """
+        self.defer(
+            timeout=self.execution_timeout,
+            trigger=SparkSubmitTrigger(
+                task_id=self.task_id,
+                polling_period_seconds=self.poll_interval,
+                application=self._application,
+                conf=self._conf,
+                conn_id=self._conn_id,
+                files=self._files,
+                py_files=self._py_files,
+                archives=self._archives,
+                driver_class_path=self._driver_class_path,
+                jars=self._jars,
+                java_class=self._java_class,
+                packages=self._packages,
+                exclude_packages=self._exclude_packages,
+                repositories=self._repositories,
+                total_executor_cores=self._total_executor_cores,
+                executor_cores=self._executor_cores,
+                executor_memory=self._executor_memory,
+                driver_memory=self._driver_memory,
+                keytab=self._keytab,
+                principal=self._principal,
+                proxy_user=self._proxy_user,
+                name=self._name,
+                num_executors=self._num_executors,
+                application_args=self._application_args,
+                env_vars=self._env_vars,
+                verbose=self._verbose,
+                spark_binary=self._spark_binary,
+            ),
+            method_name="execute_complete",
+        )
+
+    def execute_complete(
+        self, context: Dict[str, Any], event: Optional[Dict[str, Union[str, List[str]]]] = None
+    ) -> None:
+        """
+        Callback for when the trigger fires - returns immediately.
+        Relies on trigger to throw an exception, otherwise it assumes execution was
+        successful.
+        """
+        if event:
+            if event["status"] == "ERROR":
+                raise AirflowException(event["message"])
+        self.log.info("%s completed successfully.", self.task_id)
+        return None
```

Comment:
Is this a convention we follow across our code to have an explicit `return None` at the end of a function? Python by default returns None if not specified.